twilight-model/src/guild/invite/invite_type.rs

```diff
@@ -0,0 +1,66 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(from = "u8", into = "u8")]
+pub enum InviteType {
+    Guild,
+    GroupDm,
+    Friend,
+    Unknown(u8),
+}
+
+impl From<u8> for InviteType {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Self::Guild,
+            1 => Self::GroupDm,
+            2 => Self::Friend,
+            unknown => Self::Unknown(unknown),
+        }
+    }
+}
+
+impl From<InviteType> for u8 {
+    fn from(value: InviteType) -> Self {
+        match value {
+            InviteType::Guild => 0,
+            InviteType::GroupDm => 1,
+            InviteType::Friend => 2,
+            InviteType::Unknown(unknown) => unknown,
+        }
+    }
+}
+
+impl InviteType {
+    pub const fn name(&self) -> &str {
+        match self {
+            Self::Guild => "Guild",
+            Self::GroupDm => "Group",
+            Self::Friend => "Friend",
```

Comment:
Lets use the same name as the docs:
```suggestion
            Self::Guild => "GUILD",
            Self::GroupDm => "GROUP_DM",
            Self::Friend => "FRIEND",
```

Reply:
correct me if im wrong, but i dont think this is what we do for other enums having this method

Reply:
Oh yeah, you are right, I think this is something we may want to change in the future, but let us skip it here.